npm/private/utils.bzl

```diff
@@ -49,16 +56,35 @@ def _parse_pnpm_name(pnpmName):
         fail(msg)
     return (segments[0], segments[1])
 
-def _parse_pnpm_lock(lockfile_content):
-    """Parse a pnpm lock file.
+def _parse_pnpm_lock(content):
+    """Parse the content of a pnpm-lock.yaml file.s
 
     Args:
-        lockfile_content: yaml lockfile content
+        content: lockfile content
 
     Returns:
-        dict containing parsed lockfile
+        A tuple of (importers dict, packages dict)
     """
-    return _parse_yaml(lockfile_content)
+    parsed = _parse_yaml(content)
+
+    if not types.is_dict(parsed):
+        fail("lockfile should be a starlark dict")
```

Comment:
When does this happen? Is there an unhandled parser case I could fix?

Reply:
I believe this is just a defence in depth; I haven't seen this hit

Reply:
(this line was just moved around in this PR)